src/components/modals/AddDulModal.js

```diff
@@ -0,0 +1,322 @@
+    }
+    if (response === true) {
+      return true;
+    } else {
+      this.handleErrors(response);
+    }
+    return false;
+  };
+
+
+  handleErrors(message) {
+    let errMessage = '';
+    let errorTitle = 'Conflicts to resolve!';
+    if (message.indexOf("PRIMARY") > -1) {
+      errMessage = "There is a Data Use Limitation already registered with this Consent Id. ";
+    } else if (message.indexOf("name") > -1) {
+      errMessage = "There is a Data Use Limitation already registered with this name.";
+    } else if (message.indexOf("Unable to process JSON") > -1) {
+      errMessage = "Structured Limitations or Data Use has invalid format. Please write it as valid JSON.";
+    } else {
+      errorTitle = "Error, unable to create a new Data Use Limitation! ";
+      errMessage = message;
+    }
+
+    this.setState(prev => {
+      prev.disableOkBtn = true;
+      prev.error.title = errorTitle;
+      prev.error.show = true;
+      prev.error.msg = errMessage;
+      return prev;
+    });
+  };
+
+  handleChange = (changeEvent) => {
+    const name = changeEvent.target.name;
+    const value = changeEvent.target.value;
+
+    this.setState(prev => {
+      prev.disableOkBtn = value === '';
+      prev.consent[name] = value;
+      return prev;
+    });
+
+  };
+
+  onFileChange = (e) => {
+    if (e.target.files !== undefined && e.target.files[0]) {
+      let file = e.target.files[0];
+      this.setState({
+        file: file,
+      });
+    }
+  };
+
+  isValidJson = (obj, error) => {
+    try {
+      JSON.parse(obj);
+      this.setState({ disableOkBtn: true });
+      return true;
+    } catch (err) {
+      this.handleErrors(error);
+      return false;
+    }
+  };
+
+  render() {
+
+    const file = {
+      name: "MyFile.txt"
+    };
+
+    return (
+
+      BaseModal({
+        id: "addDulModal",
+        disableOkBtn: this.state.file === '' || this.state.disableOkBtn,
+        showModal: this.props.showModal,
+        onRequestClose: this.closeHandler,
+        onAfterOpen: this.afterOpenHandler,
+        imgSrc: "/images/icon_add_dul.png",
+        color: "dul",
+        title: this.state.isEditMode ? "Edit Data Use Limitations" : "Add Data Use Limitations",
+        description: this.state.isEditMode ? "Edit a Data Use Limitations Record" : "Catalog a Data Use Limitation Record",
+        action: {
+          label: this.state.isEditMode ? "Edit" : "Add",
+          handler: this.OKHandler
+        }
+      },
+        [
+
+          form({
+            className: "form-horizontal css-form",
+            name: "consentForm",
+            noValidate: true,
+            encType: "multipart/form-data"
+          }, [
+              div({ className: "form-group first-form-group" }, [
+                label({
+                  id: "lbl_consentId",
+                  className: "col-lg-3 col-md-3 col-sm-3 col-xs-4 control-label dul-color"
+                }, ["Unique id"]),
```

Comment:
nit: I would capitalize id here
```suggestion
                }, ["Unique ID"]),
```

Reply:
ok, added to coding style rules.